### DISCUSSION TITLE
Nested Aspire Projects !!

### DISCUSSION THREAD

**Discussion by 4mrit** [microsoft/aspire#4230] [Q&A]
I am currently new to dotnet and Aspire. Is there a way to add a aspire project inside a aspire project? Or I do not need to do that?
Currently I am working on a project where i am building for a Company trying to sell Hospital Management System !! I have divided project into two layer 
Parent Layer - thms ( tenant and hospital management system)
Child Layer - hms ( hospital Management System )

I have Project Setup as Follows
  ~/work/thms
    src
   │  Identity.API
   │  Media.API
   │  Tenant.API
   │  WebApp
   │  hms
   │ │  hms.ApiGateway                
   │ │  hms.AppHost                   
   │ │  hms.ServiceDefaults
   │ └  hms.sln
   │  thms.AppHost                    
   └  thms.ServiceDefaults
   * Directory.Packages.props
    hms.sln

Any Feed Back will be Appreciated !!

**Comment by davidfowl** [comment]
We're always looking for feedback from new patterns. Why do you want to create multiple app hosts? What problem does it solve for you?

**Reply by 4mrit** [reply]
I want to create multiple app hosts because i am working on SAAS for clinic management system. Thus i want to manage two projects
being Clinic Management System and Tenant Management System ( for managing customers/clinics ) where TMS is the parent project
and CMS is the child project where each project is an aspire project ! I want to be able to manage or link the child project to parent project dashboard without having to link all the projects inside the child project and Achieve project flow like this !

Parent Project -> parent project micro-services & child project
Child Project -> child project micro-services

But since i cannot add another aspire project as a project in parent Aspire Project , I cannot achieve this desired behavior .

**Reply by davidfowl** [reply]
This isn't a feature of the apphost. There's no resource hierarchy in the dashboard.

**Comment by wmeints** [comment]
I talked about this to a colleague and he mentioned a scenario like this:

When your environment grows, you typically have more than 10 microservices running. You don't want to run those on your local dev environment; instead, you want to run a group of related services on your dev environment. For DTAP, however, you want to deploy all microservices. It may be useful to have a nested structure for this case.

I can also imagine creating a CI pipeline for each group of services and integrating the various groups via virtual network peering in Azure. In this case you have more flexibility in where you store code for groups of services and how/when you want to deploy.

**Reply by davidfowl** [reply]
> DTAP

What's DTAP?


You can have as many apphost projects as you see fit for your architecture, as long as you understand how to use them for the goal you are trying to achieve. Is there any code sharing across those services?

> I can also imagine creating a CI pipeline for each group of services and integrating the various groups via virtual network peering in Azure. In this case you have more flexibility in where you store code for groups of services and how/when you want to deploy.

I'm not sure the 2 are related, but I might be missing something.

Also remember that we're at version 1, and lots of new features will be designed that may reduce the need for some pattern depending on what you use it for.

**Comment by wmeints** [comment]
DTAP: Development, Test, Acceptance, Production. I mistakenly assumed more people would use that term for their staging and production environments. For us, we typically have environment where you have multiple groups of services and resources that stick together. You can have a microservice with an associated database for auth that considered a group. And another set of microservices for ordering that have their own databases and storage. They typically don't share code. But I'm not sure if that's the same for the original poster.

Given this sort of environment, I can imagine that a nested host project. One overall structure where the root combines everything for deployment to production. And then several other app hosts for those separate groups so you can easily work on your local dev box.
